File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        if(nums == null || nums.length == 0){
            return -1;
        }

        int cnt = 1;
        int i = 1;
        int ele = nums[0];
        int n = nums.length;

        while(i < n){
            if(cnt == 0){
                ele = nums[i];
            }
            if(nums[i] == ele){
                cnt++;
            } else {
                cnt--;
            }

            i++;
        }

        for(i = 0 ; i < n ; i++){
            if(nums[i] == ele){
                cnt++;
            }
        }

        return (cnt > n/2) ? ele : -1;
        
    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        majority = nums[0]
        count = 0
        for i in range(len(nums)):
            if nums[i] == majority:
                count = count+1
            if nums[i] != majority:
                count = count-1
            if count == 0:
                majority = nums[i]
                count = count+1
        
        return majority
        